{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File exists.\n",
      "CCLE data head:               gene_id                                     transcript_ids  \\\n",
      "0  ENSG00000000003.10  ENST00000373020.4,ENST00000494424.1,ENST000004...   \n",
      "1   ENSG00000000005.5                ENST00000373031.4,ENST00000485971.1   \n",
      "2   ENSG00000000419.8  ENST00000371582.4,ENST00000371583.5,ENST000003...   \n",
      "3   ENSG00000000457.9  ENST00000367770.1,ENST00000367771.6,ENST000003...   \n",
      "4  ENSG00000000460.12  ENST00000286031.6,ENST00000359326.4,ENST000004...   \n",
      "\n",
      "   22RV1_PROSTATE  2313287_STOMACH  253JBV_URINARY_TRACT  253J_URINARY_TRACT  \\\n",
      "0            5.28             7.01                 22.80               22.88   \n",
      "1            0.00             0.00                  0.00                0.00   \n",
      "2           73.38           108.99                 56.51               45.39   \n",
      "3            9.76            16.76                  2.58                3.25   \n",
      "4           24.51            13.32                 10.86                5.26   \n",
      "\n",
      "   42MGBA_CENTRAL_NERVOUS_SYSTEM  5637_URINARY_TRACT  59M_OVARY  \\\n",
      "0                          23.09               57.94      16.30   \n",
      "1                           0.00                0.00       0.00   \n",
      "2                          99.28               98.95      79.68   \n",
      "3                           2.73                3.11       2.69   \n",
      "4                           9.27               31.61       9.62   \n",
      "\n",
      "   639V_URINARY_TRACT  ...  WSUDLCL2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
      "0               32.02  ...                                         0.74   \n",
      "1                0.00  ...                                         0.00   \n",
      "2              125.99  ...                                        41.71   \n",
      "3                2.69  ...                                         5.60   \n",
      "4               10.45  ...                                        20.21   \n",
      "\n",
      "   YAPC_PANCREAS  YD10B_UPPER_AERODIGESTIVE_TRACT  YD15_SALIVARY_GLAND  \\\n",
      "0          50.29                            14.18                24.13   \n",
      "1           0.00                             0.00                 0.00   \n",
      "2         103.70                            97.07               104.97   \n",
      "3           7.76                             2.96                 5.84   \n",
      "4          12.84                            11.38                14.58   \n",
      "\n",
      "   YD38_UPPER_AERODIGESTIVE_TRACT  YD8_UPPER_AERODIGESTIVE_TRACT  \\\n",
      "0                           20.94                          47.96   \n",
      "1                            0.00                           0.00   \n",
      "2                           96.86                         119.50   \n",
      "3                            4.17                           5.06   \n",
      "4                           22.13                           9.58   \n",
      "\n",
      "   YH13_CENTRAL_NERVOUS_SYSTEM  YKG1_CENTRAL_NERVOUS_SYSTEM  ZR751_BREAST  \\\n",
      "0                        28.92                        61.08          8.12   \n",
      "1                         0.00                         0.00          0.00   \n",
      "2                        64.09                       109.13        139.23   \n",
      "3                         4.94                         5.91         15.96   \n",
      "4                        13.35                        17.40         17.45   \n",
      "\n",
      "   ZR7530_BREAST  \n",
      "0          16.81  \n",
      "1           0.00  \n",
      "2          56.20  \n",
      "3          13.76  \n",
      "4          11.53  \n",
      "\n",
      "[5 rows x 1021 columns]\n"
     ]
    }
   ],
   "source": [
    "# read CCLE\n",
    "\n",
    "import pandas as pd\n",
    "import os\n",
    "import gzip\n",
    "\n",
    "\n",
    "# CCLE has the gene expression\n",
    "\n",
    "# File path\n",
    "folder = \"./datasets/\"\n",
    "ccle_path = os.path.join(folder, \"CCLE_RNAseq_rsem_genes_tpm_20180929.txt.gz\")\n",
    "\n",
    "\n",
    "if not os.path.exists(ccle_path):\n",
    "    print(\"File not found. Please check the file path.\")\n",
    "else:\n",
    "    print(\"File exists.\")\n",
    "\n",
    "# Read the gzipped file\n",
    "ccle_data = pd.read_csv(ccle_path, compression='gzip', sep='\\t', comment='#')\n",
    "\n",
    "# Print the head of the dataset\n",
    "print(\"CCLE data head:\", ccle_data.head())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gencode file exists.\n",
      "Gencode data head:    chr1   HAVANA        gene  11869  14412  .  + ..1  \\\n",
      "0  chr1   HAVANA  transcript  11869  14409  .  +   .   \n",
      "1  chr1   HAVANA        exon  11869  12227  .  +   .   \n",
      "2  chr1   HAVANA        exon  12613  12721  .  +   .   \n",
      "3  chr1   HAVANA        exon  13221  14409  .  +   .   \n",
      "4  chr1  ENSEMBL  transcript  11872  14412  .  +   .   \n",
      "5  chr1  ENSEMBL        exon  11872  12227  .  +   .   \n",
      "6  chr1  ENSEMBL        exon  12613  12721  .  +   .   \n",
      "7  chr1  ENSEMBL        exon  13225  14412  .  +   .   \n",
      "8  chr1  ENSEMBL  transcript  11874  14409  .  +   .   \n",
      "9  chr1  ENSEMBL        exon  11874  12227  .  +   .   \n",
      "\n",
      "  gene_id \"ENSG00000223972.4\"; transcript_id \"ENSG00000223972.4\"; gene_type \"pseudogene\"; gene_status \"KNOWN\"; gene_name \"DDX11L1\"; transcript_type \"pseudogene\"; transcript_status \"KNOWN\"; transcript_name \"DDX11L1\"; level 2; havana_gene \"OTTHUMG00000000961.2\";  \n",
      "0  gene_id \"ENSG00000223972.4\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
      "1  gene_id \"ENSG00000223972.4\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
      "2  gene_id \"ENSG00000223972.4\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
      "3  gene_id \"ENSG00000223972.4\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
      "4  gene_id \"ENSG00000223972.4\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
      "5  gene_id \"ENSG00000223972.4\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
      "6  gene_id \"ENSG00000223972.4\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
      "7  gene_id \"ENSG00000223972.4\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
      "8  gene_id \"ENSG00000223972.4\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
      "9  gene_id \"ENSG00000223972.4\"; transcript_id \"EN...                                                                                                                                                                                                                  \n"
     ]
    }
   ],
   "source": [
    "# transform ENSENBL IDs into official gene symbols\n",
    "\n",
    "# read gene annotation\n",
    "\n",
    "# File path\n",
    "gencode_path = os.path.join(folder, \"gencode.v19.annotation.gtf.gz\")\n",
    "\n",
    "if not os.path.exists(gencode_path):\n",
    "    print(\"Gencode file not found. Please check the file path.\")\n",
    "else:\n",
    "    print(\"Gencode file exists.\")\n",
    "\n",
    "# Read the gzipped file\n",
    "gencode_data = pd.read_csv(gencode_path, compression='gzip', sep='\\t', comment='#')\n",
    "\n",
    "# Print the head of the dataset\n",
    "print(\"Gencode data head:\", gencode_data.head(10))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENSG00000223972.4: DDX11L1\n",
      "ENSG00000227232.4: WASH7P\n",
      "ENSG00000243485.2: MIR1302-11\n",
      "ENSG00000237613.2: FAM138A\n",
      "ENSG00000268020.2: OR4G4P\n",
      "ENSG00000240361.1: OR4G11P\n",
      "ENSG00000186092.4: OR4F5\n",
      "ENSG00000238009.2: RP11-34P13.7\n",
      "ENSG00000239945.1: RP11-34P13.8\n",
      "ENSG00000233750.3: CICP27\n"
     ]
    }
   ],
   "source": [
    "# transform ENSENBL IDs into official gene symbols\n",
    "\n",
    "\n",
    "# Read the GTF file (only necessary columns for gene_id and gene_name)\n",
    "with gzip.open(gencode_path, 'rt') as f:\n",
    "    # Parse only rows containing \"gene\" and extract the relevant column\n",
    "    gtf_data = pd.read_csv(f, sep='\\t', comment='#', header=None, usecols=[8], names=['attributes'])\n",
    "\n",
    "# Extract gene_id and gene_name from the attributes column\n",
    "def parse_attributes(attributes):\n",
    "    \"\"\"Extract gene_id and gene_name from the attributes string.\"\"\"\n",
    "    fields = attributes.split(';')\n",
    "    gene_id = next((field.split('\"')[1] for field in fields if field.strip().startswith(\"gene_id\")), None)\n",
    "    gene_name = next((field.split('\"')[1] for field in fields if field.strip().startswith(\"gene_name\")), None)\n",
    "    return gene_id, gene_name\n",
    "\n",
    "# Apply the parsing function to extract gene_id and gene_name\n",
    "mapping = gtf_data['attributes'].apply(parse_attributes)\n",
    "gene_mapping_df = pd.DataFrame(mapping.tolist(), columns=['gene_id', 'gene_name'])\n",
    "\n",
    "# Create a dictionary for mapping\n",
    "gene_mapping = dict(zip(gene_mapping_df['gene_id'], gene_mapping_df['gene_name']))\n",
    "\n",
    "# Print the first 10 items of the dictionary\n",
    "for i, (key, value) in enumerate(gene_mapping.items()):\n",
    "    print(f\"{key}: {value}\")\n",
    "    if i == 9:  # Stop after printing 10 items\n",
    "        break\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed CCLE data     gene_id                                     transcript_ids  \\\n",
      "0    TSPAN6  ENST00000373020.4,ENST00000494424.1,ENST000004...   \n",
      "1      TNMD                ENST00000373031.4,ENST00000485971.1   \n",
      "2      DPM1  ENST00000371582.4,ENST00000371583.5,ENST000003...   \n",
      "3     SCYL3  ENST00000367770.1,ENST00000367771.6,ENST000003...   \n",
      "4  C1orf112  ENST00000286031.6,ENST00000359326.4,ENST000004...   \n",
      "\n",
      "   22RV1_PROSTATE  2313287_STOMACH  253JBV_URINARY_TRACT  253J_URINARY_TRACT  \\\n",
      "0            5.28             7.01                 22.80               22.88   \n",
      "1            0.00             0.00                  0.00                0.00   \n",
      "2           73.38           108.99                 56.51               45.39   \n",
      "3            9.76            16.76                  2.58                3.25   \n",
      "4           24.51            13.32                 10.86                5.26   \n",
      "\n",
      "   42MGBA_CENTRAL_NERVOUS_SYSTEM  5637_URINARY_TRACT  59M_OVARY  \\\n",
      "0                          23.09               57.94      16.30   \n",
      "1                           0.00                0.00       0.00   \n",
      "2                          99.28               98.95      79.68   \n",
      "3                           2.73                3.11       2.69   \n",
      "4                           9.27               31.61       9.62   \n",
      "\n",
      "   639V_URINARY_TRACT  ...  WSUDLCL2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  \\\n",
      "0               32.02  ...                                         0.74   \n",
      "1                0.00  ...                                         0.00   \n",
      "2              125.99  ...                                        41.71   \n",
      "3                2.69  ...                                         5.60   \n",
      "4               10.45  ...                                        20.21   \n",
      "\n",
      "   YAPC_PANCREAS  YD10B_UPPER_AERODIGESTIVE_TRACT  YD15_SALIVARY_GLAND  \\\n",
      "0          50.29                            14.18                24.13   \n",
      "1           0.00                             0.00                 0.00   \n",
      "2         103.70                            97.07               104.97   \n",
      "3           7.76                             2.96                 5.84   \n",
      "4          12.84                            11.38                14.58   \n",
      "\n",
      "   YD38_UPPER_AERODIGESTIVE_TRACT  YD8_UPPER_AERODIGESTIVE_TRACT  \\\n",
      "0                           20.94                          47.96   \n",
      "1                            0.00                           0.00   \n",
      "2                           96.86                         119.50   \n",
      "3                            4.17                           5.06   \n",
      "4                           22.13                           9.58   \n",
      "\n",
      "   YH13_CENTRAL_NERVOUS_SYSTEM  YKG1_CENTRAL_NERVOUS_SYSTEM  ZR751_BREAST  \\\n",
      "0                        28.92                        61.08          8.12   \n",
      "1                         0.00                         0.00          0.00   \n",
      "2                        64.09                       109.13        139.23   \n",
      "3                         4.94                         5.91         15.96   \n",
      "4                        13.35                        17.40         17.45   \n",
      "\n",
      "   ZR7530_BREAST  \n",
      "0          16.81  \n",
      "1           0.00  \n",
      "2          56.20  \n",
      "3          13.76  \n",
      "4          11.53  \n",
      "\n",
      "[5 rows x 1021 columns]\n",
      "(57820, 1021)\n"
     ]
    }
   ],
   "source": [
    "# Replace gene IDs with gene names using the mapping dictionary\n",
    "ccle_data['gene_id'] = ccle_data['gene_id'].map(gene_mapping)\n",
    "\n",
    "\n",
    "# Display the first few rows of the transformed DataFrame\n",
    "print(\"Transformed CCLE data\", ccle_data.head())\n",
    "print(ccle_data.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    gene_id  22RV1_PROSTATE  2313287_STOMACH  253JBV_URINARY_TRACT  \\\n",
      "0   5S_rRNA        0.000000         0.000000              0.001429   \n",
      "1       7SK        0.121429         0.131429              0.075714   \n",
      "2      A1BG        2.300000         0.140000              0.430000   \n",
      "3  A1BG-AS1        6.480000         0.120000              0.340000   \n",
      "4      A1CF        8.590000         3.470000              0.030000   \n",
      "\n",
      "   253J_URINARY_TRACT  42MGBA_CENTRAL_NERVOUS_SYSTEM  5637_URINARY_TRACT  \\\n",
      "0            0.004286                       0.015714                0.00   \n",
      "1            0.098571                       0.004286                0.03   \n",
      "2            1.060000                      49.810000                0.18   \n",
      "3            1.100000                      16.790000                0.14   \n",
      "4            0.060000                       0.030000                0.00   \n",
      "\n",
      "   59M_OVARY  639V_URINARY_TRACT  647V_URINARY_TRACT  ...  \\\n",
      "0   0.011429            0.015714            0.002857  ...   \n",
      "1   0.082857            0.161429            0.004286  ...   \n",
      "2  30.810000           31.300000            1.520000  ...   \n",
      "3   2.200000            7.330000            0.630000  ...   \n",
      "4   0.010000            0.000000            0.000000  ...   \n",
      "\n",
      "   WSUDLCL2_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  YAPC_PANCREAS  \\\n",
      "0                                     0.000000       0.000000   \n",
      "1                                     1.551429       0.054286   \n",
      "2                                    73.590000       0.000000   \n",
      "3                                     6.930000       0.140000   \n",
      "4                                     0.000000       0.020000   \n",
      "\n",
      "   YD10B_UPPER_AERODIGESTIVE_TRACT  YD15_SALIVARY_GLAND  \\\n",
      "0                         0.081429             0.008571   \n",
      "1                         0.384286             0.051429   \n",
      "2                         0.000000             0.610000   \n",
      "3                         0.690000             0.500000   \n",
      "4                         0.260000             0.000000   \n",
      "\n",
      "   YD38_UPPER_AERODIGESTIVE_TRACT  YD8_UPPER_AERODIGESTIVE_TRACT  \\\n",
      "0                        0.000000                       0.004286   \n",
      "1                        0.004286                       0.567143   \n",
      "2                        0.340000                       1.200000   \n",
      "3                        0.110000                       4.330000   \n",
      "4                        0.000000                       0.010000   \n",
      "\n",
      "   YH13_CENTRAL_NERVOUS_SYSTEM  YKG1_CENTRAL_NERVOUS_SYSTEM  ZR751_BREAST  \\\n",
      "0                     0.000000                     0.012857      0.000000   \n",
      "1                     0.037143                     0.040000      0.171429   \n",
      "2                    27.570000                    21.040000     47.740000   \n",
      "3                    13.720000                     5.960000      9.000000   \n",
      "4                     0.010000                     0.000000      0.060000   \n",
      "\n",
      "   ZR7530_BREAST  \n",
      "0       0.000000  \n",
      "1       0.034286  \n",
      "2      33.130000  \n",
      "3       7.080000  \n",
      "4       0.000000  \n",
      "\n",
      "[5 rows x 1020 columns]\n",
      "(55765, 1020)\n"
     ]
    }
   ],
   "source": [
    "# Average duplicates \n",
    "\n",
    "# Group by the 'gene_id' column (now containing gene names)\n",
    "# and calculate the mean for all other columns\n",
    "averaged_data = ccle_data.groupby('gene_id', as_index=False).mean(numeric_only=True)\n",
    "\n",
    "# Display the first few rows of the resulting DataFrame\n",
    "print(averaged_data.head())\n",
    "print(averaged_data.shape)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File exists.\n",
      "  DATASET  NLME_RESULT_ID  NLME_CURVE_ID  COSMIC_ID CELL_LINE_NAME  \\\n",
      "0   GDSC1             261       12258263     683665         MC-CAR   \n",
      "1   GDSC1             261       12259075     684055            ES3   \n",
      "2   GDSC1             261       12259376     684057            ES5   \n",
      "3   GDSC1             261       12259663     684059            ES7   \n",
      "4   GDSC1             261       12259967     684062          EW-11   \n",
      "5   GDSC1             261       12260226     684072        SK-ES-1   \n",
      "6   GDSC1             261       12260526     684681      NCI-H1395   \n",
      "7   GDSC1             261       12260833     687448       COLO-829   \n",
      "8   GDSC1             261       12263174     687562        8-MG-BA   \n",
      "9   GDSC1             261       12263728     687568           GB-1   \n",
      "\n",
      "  SANGER_MODEL_ID     TCGA_DESC  DRUG_ID  DRUG_NAME PUTATIVE_TARGET  \\\n",
      "0       SIDM00636            MM        1  Erlotinib            EGFR   \n",
      "1       SIDM00265  UNCLASSIFIED        1  Erlotinib            EGFR   \n",
      "2       SIDM00263  UNCLASSIFIED        1  Erlotinib            EGFR   \n",
      "3       SIDM00269  UNCLASSIFIED        1  Erlotinib            EGFR   \n",
      "4       SIDM00203  UNCLASSIFIED        1  Erlotinib            EGFR   \n",
      "5       SIDM01111  UNCLASSIFIED        1  Erlotinib            EGFR   \n",
      "6       SIDM00644          LUAD        1  Erlotinib            EGFR   \n",
      "7       SIDM00909          SKCM        1  Erlotinib            EGFR   \n",
      "8       SIDM00998           GBM        1  Erlotinib            EGFR   \n",
      "9       SIDM00581           GBM        1  Erlotinib            EGFR   \n",
      "\n",
      "     PATHWAY_NAME  COMPANY_ID WEBRELEASE  MIN_CONC  MAX_CONC   LN_IC50  \\\n",
      "0  EGFR signaling        1045          Y  0.007813       2.0  2.395838   \n",
      "1  EGFR signaling        1045          Y  0.007813       2.0  3.141303   \n",
      "2  EGFR signaling        1045          Y  0.007813       2.0  3.969070   \n",
      "3  EGFR signaling        1045          Y  0.007813       2.0  2.692884   \n",
      "4  EGFR signaling        1045          Y  0.007813       2.0  2.478795   \n",
      "5  EGFR signaling        1045          Y  0.007813       2.0  2.034131   \n",
      "6  EGFR signaling        1045          Y  0.007813       2.0  3.987058   \n",
      "7  EGFR signaling        1045          Y  0.007813       2.0  2.967109   \n",
      "8  EGFR signaling        1045          Y  0.007813       2.0  2.494990   \n",
      "9  EGFR signaling        1045          Y  0.007813       2.0  3.605185   \n",
      "\n",
      "        AUC      RMSE   Z_SCORE  \n",
      "0  0.982116  0.022518 -0.192056  \n",
      "1  0.984820  0.031831  0.505823  \n",
      "2  0.985696  0.026047  1.280750  \n",
      "3  0.972701  0.110056  0.086028  \n",
      "4  0.944463  0.087010 -0.114395  \n",
      "5  0.950763  0.016288 -0.530674  \n",
      "6  0.981350  0.032070  1.297590  \n",
      "7  0.954785  0.180250  0.342749  \n",
      "8  0.980213  0.041884 -0.099233  \n",
      "9  0.984637  0.069284  0.940093  \n"
     ]
    }
   ],
   "source": [
    "# read GDSC 1\n",
    "\n",
    "# File path\n",
    "folder2 = \"./datasets_OG/\"\n",
    "gdsc1_path = os.path.join(folder2, \"GDSC1_fitted_dose_response_17Jul19.csv\")\n",
    "\n",
    "if not os.path.exists(gdsc1_path):\n",
    "    print(\"File not found. Please check the file path.\")\n",
    "else:\n",
    "    print(\"File exists.\")\n",
    "\n",
    "gdsc1_data = pd.read_csv(gdsc1_path)\n",
    "\n",
    "# Print the head of the dataset\n",
    "print(gdsc1_data.head(10))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File exists.\n",
      "  DATASET  NLME_RESULT_ID  NLME_CURVE_ID  COSMIC_ID CELL_LINE_NAME  \\\n",
      "0   GDSC2             290       14789087     909751           SW48   \n",
      "1   GDSC2             290       14789101     905962          SW620   \n",
      "2   GDSC2             290       14789115     910927         CAL-51   \n",
      "3   GDSC2             290       14789148     908442       NCI-H508   \n",
      "4   GDSC2             290       14789169     909740        SNU-C2B   \n",
      "5   GDSC2             290       14789183     909718        SK-CO-1   \n",
      "6   GDSC2             290       14789197     905971        HCC2998   \n",
      "7   GDSC2             290       14789247     749712        HCC1395   \n",
      "8   GDSC2             290       14789284     908122     MDA-MB-453   \n",
      "9   GDSC2             290       14789317     910852       CAL-85-1   \n",
      "\n",
      "  SANGER_MODEL_ID TCGA_DESC  DRUG_ID     DRUG_NAME PUTATIVE_TARGET  \\\n",
      "0       SIDM00837    COREAD     1003  Camptothecin            TOP1   \n",
      "1       SIDM00841    COREAD     1003  Camptothecin            TOP1   \n",
      "2       SIDM00933      BRCA     1003  Camptothecin            TOP1   \n",
      "3       SIDM00777    COREAD     1003  Camptothecin            TOP1   \n",
      "4       SIDM00781    COREAD     1003  Camptothecin            TOP1   \n",
      "5       SIDM01096    COREAD     1003  Camptothecin            TOP1   \n",
      "6       SIDM00118    COREAD     1003  Camptothecin            TOP1   \n",
      "7       SIDM00884      BRCA     1003  Camptothecin            TOP1   \n",
      "8       SIDM00272      BRCA     1003  Camptothecin            TOP1   \n",
      "9       SIDM00928      BRCA     1003  Camptothecin            TOP1   \n",
      "\n",
      "      PATHWAY_NAME  COMPANY_ID WEBRELEASE  MIN_CONC  MAX_CONC   LN_IC50  \\\n",
      "0  DNA replication        1046          Y  0.000098       0.1 -3.784746   \n",
      "1  DNA replication        1046          Y  0.000098       0.1 -3.092858   \n",
      "2  DNA replication        1046          Y  0.000098       0.1 -3.521093   \n",
      "3  DNA replication        1046          Y  0.000098       0.1 -3.447166   \n",
      "4  DNA replication        1046          Y  0.000098       0.1  0.067435   \n",
      "5  DNA replication        1046          Y  0.000098       0.1 -5.331648   \n",
      "6  DNA replication        1046          Y  0.000098       0.1 -1.265275   \n",
      "7  DNA replication        1046          Y  0.000098       0.1 -2.309078   \n",
      "8  DNA replication        1046          Y  0.000098       0.1 -2.014633   \n",
      "9  DNA replication        1046          Y  0.000098       0.1 -1.177570   \n",
      "\n",
      "        AUC      RMSE   Z_SCORE  \n",
      "0  0.721202  0.111265 -0.815291  \n",
      "1  0.783543  0.097285 -0.478141  \n",
      "2  0.757702  0.084846 -0.686816  \n",
      "3  0.769315  0.073790 -0.650792  \n",
      "4  0.928017  0.079673  1.061842  \n",
      "5  0.555392  0.127926 -1.569083  \n",
      "6  0.955958  0.099446  0.412424  \n",
      "7  0.867832  0.107282 -0.096212  \n",
      "8  0.887863  0.075300  0.047269  \n",
      "9  0.932126  0.064493  0.455162  \n"
     ]
    }
   ],
   "source": [
    "# read GDSC 2\n",
    "\n",
    "# File path\n",
    "gdsc2_path = os.path.join(folder2, \"GDSC2_fitted_dose_response_15Oct19.csv\")\n",
    "\n",
    "if not os.path.exists(gdsc2_path):\n",
    "    print(\"File not found. Please check the file path.\")\n",
    "else:\n",
    "    print(\"File exists.\")\n",
    "\n",
    "gdsc2_data = pd.read_csv(gdsc2_path)\n",
    "\n",
    "# Print the head of the dataset\n",
    "print(gdsc2_data.head(10))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  DATASET  NLME_RESULT_ID  NLME_CURVE_ID  COSMIC_ID CELL_LINE_NAME  \\\n",
      "0   GDSC1             261       12258263     683665         MC-CAR   \n",
      "1   GDSC1             261       12259075     684055            ES3   \n",
      "2   GDSC1             261       12259376     684057            ES5   \n",
      "3   GDSC1             261       12259663     684059            ES7   \n",
      "4   GDSC1             261       12259967     684062          EW-11   \n",
      "\n",
      "  SANGER_MODEL_ID     TCGA_DESC  DRUG_ID  DRUG_NAME PUTATIVE_TARGET  \\\n",
      "0       SIDM00636            MM        1  Erlotinib            EGFR   \n",
      "1       SIDM00265  UNCLASSIFIED        1  Erlotinib            EGFR   \n",
      "2       SIDM00263  UNCLASSIFIED        1  Erlotinib            EGFR   \n",
      "3       SIDM00269  UNCLASSIFIED        1  Erlotinib            EGFR   \n",
      "4       SIDM00203  UNCLASSIFIED        1  Erlotinib            EGFR   \n",
      "\n",
      "     PATHWAY_NAME  COMPANY_ID WEBRELEASE  MIN_CONC  MAX_CONC   LN_IC50  \\\n",
      "0  EGFR signaling        1045          Y  0.007813       2.0  2.395838   \n",
      "1  EGFR signaling        1045          Y  0.007813       2.0  3.141303   \n",
      "2  EGFR signaling        1045          Y  0.007813       2.0  3.969070   \n",
      "3  EGFR signaling        1045          Y  0.007813       2.0  2.692884   \n",
      "4  EGFR signaling        1045          Y  0.007813       2.0  2.478795   \n",
      "\n",
      "        AUC      RMSE   Z_SCORE  \n",
      "0  0.982116  0.022518 -0.192056  \n",
      "1  0.984820  0.031831  0.505823  \n",
      "2  0.985696  0.026047  1.280750  \n",
      "3  0.972701  0.110056  0.086028  \n",
      "4  0.944463  0.087010 -0.114395  \n"
     ]
    }
   ],
   "source": [
    "# Ensure both files have the same column names (if needed)\n",
    "assert list(gdsc1_data.columns) == list(gdsc2_data.columns), \"Column names must match!\"\n",
    "\n",
    "# Drop duplicates from the first file if they exist in the second file\n",
    "# Keeping rows from df2 when duplicates are found\n",
    "gdsc1_filtered = gdsc1_data[~gdsc1_data[['CELL_LINE_NAME', 'DRUG_ID']].isin(gdsc2_data[['CELL_LINE_NAME', 'DRUG_ID']].to_dict(orient='list')).all(axis=1)]\n",
    "\n",
    "# Combine the filtered rows from df1 with all rows from df2\n",
    "combined_gdsc_df = pd.concat([gdsc1_filtered, gdsc2_data], ignore_index=True)\n",
    "\n",
    "# Display the result\n",
    "print(combined_gdsc_df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of rows: 353893\n",
      "Number of columns: 19\n"
     ]
    }
   ],
   "source": [
    "# Check the shape of the combined DataFrame\n",
    "num_rows, num_columns = combined_gdsc_df.shape\n",
    "\n",
    "print(f\"Number of rows: {num_rows}\")\n",
    "print(f\"Number of columns: {num_columns}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create data matrix and IC50 vector for each drug\n",
    "\n",
    "def create_matrix_and_ic50_for_drug(drug_id, combined_gdsc_df, averaged_data):\n",
    "    \"\"\"\n",
    "    Create gene expression data matrix and IC50 vector for a specific DRUG_ID.\n",
    "\n",
    "    Parameters:\n",
    "        drug_id (int): The DRUG_ID for which to create the matrix and vector.\n",
    "        combined_gdsc_df (pd.DataFrame): GDSC dataset with DRUG_ID, CELL_LINE_NAME, and LN_IC50 columns.\n",
    "        averaged_data (pd.DataFrame): CCLE dataset with gene expression profiles as rows and cell line columns.\n",
    "\n",
    "    Returns:\n",
    "        tuple: (gene_expression_matrix, ic50_vector)\n",
    "            - gene_expression_matrix (pd.DataFrame): Gene expression matrix (G x N, CCLE).\n",
    "            - ic50_vector (list): Corresponding IC50 values (length N, CCLE).\n",
    "    \"\"\"\n",
    "    # Create mapping from CELL_LINE_ID to CCLE columns\n",
    "    ccle_columns = {col.split('_')[0]: col for col in averaged_data.columns if '_' in col}\n",
    "\n",
    "    # Filter rows for the given DRUG_ID\n",
    "    drug_data = combined_gdsc_df[combined_gdsc_df['DRUG_ID'] == drug_id]\n",
    "\n",
    "    # Initialize lists to store matrix columns and IC50 values\n",
    "    gene_expression_matrix = []\n",
    "    ic50_vector = []\n",
    "    matching_cell_lines = []\n",
    "\n",
    "    # Match CELL_LINE_NAMEs to CCLE columns\n",
    "    for cell_line in drug_data['CELL_LINE_NAME']:\n",
    "        if cell_line in ccle_columns:\n",
    "            ccle_column_name = ccle_columns[cell_line]\n",
    "\n",
    "            # Append the gene expression profile to the matrix\n",
    "            gene_expression_matrix.append(averaged_data[ccle_column_name].values)\n",
    "\n",
    "            # Append IC50 value for this cell line\n",
    "            ic50_value = float(drug_data[drug_data['CELL_LINE_NAME'] == cell_line]['LN_IC50'].values[0])\n",
    "            ic50_vector.append(ic50_value)\n",
    "            matching_cell_lines.append(cell_line)\n",
    "\n",
    "    # If no matching cell lines, return None\n",
    "    if not gene_expression_matrix:\n",
    "        print(f\"No matching cell lines found for DRUG_ID {drug_id}.\")\n",
    "        return None, None\n",
    "\n",
    "    # Convert the gene expression matrix to a DataFrame with rows as genes and columns as cell lines\n",
    "    gene_expression_matrix = pd.DataFrame(gene_expression_matrix).T\n",
    "    gene_expression_matrix.columns = matching_cell_lines\n",
    "\n",
    "    return gene_expression_matrix, ic50_vector"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[ 1  3  5  6  9 11 17 29 30 32]\n",
      "Number of unique DRUG_IDs: 453\n",
      "Drug ID with the maximum count: 1032, Count: 964\n",
      "Drug ID with the minimum count: 1086, Count: 44\n"
     ]
    }
   ],
   "source": [
    "# Get all unique drug IDs\n",
    "unique_drug_ids = combined_gdsc_df['DRUG_ID'].unique()\n",
    "print(unique_drug_ids[:10]) #first 10 unique drug IDs\n",
    "\n",
    "# Count the number of unique drug IDs\n",
    "num_unique_drugs = combined_gdsc_df['DRUG_ID'].nunique()\n",
    "print(f\"Number of unique DRUG_IDs: {num_unique_drugs}\")\n",
    "\n",
    "# Count the occurrences of each DRUG_ID\n",
    "drug_id_counts = combined_gdsc_df['DRUG_ID'].value_counts()\n",
    "\n",
    "# Find the drug IDs with the maximum and minimum counts\n",
    "max_drug_id, max_count = drug_id_counts.idxmax(), drug_id_counts.max()\n",
    "min_drug_id, min_count = drug_id_counts.idxmin(), drug_id_counts.min()\n",
    "\n",
    "print(f\"Drug ID with the maximum count: {max_drug_id}, Count: {max_count}\")\n",
    "print(f\"Drug ID with the minimum count: {min_drug_id}, Count: {min_count}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene Expression Matrix for DRUG_ID 1:\n",
      "    HCC1187   HCC1599   HCC2157   HCC2218      KELLY       SIMA       EKVX  \\\n",
      "0  0.017143  0.000000  0.000000  0.000000   0.000000   0.004286   0.014286   \n",
      "1  0.101429  0.195714  0.414286  0.087143   0.012857   0.000000   0.064286   \n",
      "2  0.580000  3.050000  1.390000  6.630000  13.630000  47.070000  16.270000   \n",
      "3  0.920000  2.070000  2.780000  1.090000   5.430000   5.460000   3.210000   \n",
      "4  0.010000  0.040000  0.020000  0.010000   0.040000   0.020000   0.020000   \n",
      "\n",
      "    LOXIMVI  SNB75      SF539  ...        CMK       CA46        RL     ST486  \\\n",
      "0  0.000000   0.00   0.000000  ...   0.000000   0.000000  0.000000  0.000000   \n",
      "1  0.017143   0.03   0.101429  ...   0.774286   2.951429  2.345714  6.711429   \n",
      "2  0.000000   0.49  15.880000  ...  14.700000  41.090000  5.780000  9.920000   \n",
      "3  0.030000   0.66   9.900000  ...   4.600000   7.510000  2.470000  5.240000   \n",
      "4  0.020000   0.00   0.020000  ...   0.000000   0.030000  0.000000  0.000000   \n",
      "\n",
      "       A101D        NB1      MDST8        EJM        NB4        JM1  \n",
      "0   0.004286   0.000000   0.000000   0.284286   0.001429   0.000000  \n",
      "1   0.068571   0.001429   0.058571   0.622857   2.642857   1.615714  \n",
      "2  35.570000  44.440000  45.220000  11.610000  12.450000  20.730000  \n",
      "3  13.300000  11.770000  13.460000   6.640000   2.710000   6.680000  \n",
      "4   0.000000   0.000000   0.030000   0.000000   0.010000   0.010000  \n",
      "\n",
      "[5 rows x 42 columns]\n",
      "\n",
      "IC50 Vector:\n",
      "[2.9598899999999997, 2.791017, 1.417929, 3.565485, 2.963196, 3.383066, -1.097496, 3.299372, 3.99508, 2.471027]\n",
      "Number of cell lines in gene_expression_matrix: 42\n",
      "Number of genes in gene_expression_matrix: 55765\n",
      "Length of ic50_vector: 42\n"
     ]
    }
   ],
   "source": [
    "# Specify a DRUG_ID\n",
    "drug_id = 1\n",
    "\n",
    "# Call the function with the specified drug ID\n",
    "gene_expression_matrix, ic50_vector = create_matrix_and_ic50_for_drug(drug_id, combined_gdsc_df, averaged_data)\n",
    "\n",
    "# Check the output\n",
    "if gene_expression_matrix is not None:\n",
    "    print(f\"Gene Expression Matrix for DRUG_ID {drug_id}:\")\n",
    "    print(gene_expression_matrix.head())\n",
    "    print(\"\\nIC50 Vector:\")\n",
    "    print(ic50_vector[:10])  # Print the first 10 values\n",
    "else:\n",
    "    print(f\"No data available for DRUG_ID {drug_id}.\")\n",
    "\n",
    "print(f\"Number of cell lines in gene_expression_matrix: {gene_expression_matrix.shape[1]}\")\n",
    "print(f\"Number of genes in gene_expression_matrix: {gene_expression_matrix.shape[0]}\")\n",
    "print(f\"Length of ic50_vector: {len(ic50_vector)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\ndrug_ids = [\\n    1, 3, 5, 6, 9, 11, 17, 29, 30, 32, 34, 35, 37, 38, 41, 45, 51, 52, 53, 54, 55, \\n    56, 59, 60, 62, 63, 64, 71, 83, 86, 87, 88, 89, 91, 94, 104, 106, 110, 111, 119, \\n    127, 133, 134, 135, 136, 140, 147, 150, 151, 152, 153, 154, 155, 156, 157, 158, \\n    159, 163, 164, 165, 166, 167, 170, 171, 172, 173, 175, 176, 177, 178, 179, 180, \\n    182, 184, 185, 186, 190, 192, 193, 194, 196, 197, 199, 200, 201, 202, 203, 204, \\n    205, 206, 207, 208, 211, 219, 221, 222, 223, 224, 225, 226, 228, 229, 230, 231, \\n    235, 236, 238, 245, 249, 252, 253, 254, 255, 256, 257, 258, 260, 261, 262, 263, \\n    264, 265, 266, 268, 269, 271, 272, 273, 274, 275, 276, 277, 279, 281, 282, 283, \\n    284, 285, 286, 287, 288, 289, 290, 291, 292, 293, 294, 295, 298, 299, 300, 301, \\n    302, 303, 304, 305, 306, 308, 309, 310, 312, 317, 326, 328, 329, 330, 331, 332, \\n    333, 341, 342, 344, 345, 346, 356, 362, 363, 366, 371, 372, 374, 375, 376, 380, \\n    381, 382, 406, 407, 408, 409, 410, 412, 415, 416, 427, 428, 431, 432, 435, 436, \\n    437, 438, 439, 442, 446, 447, 449, 461, 474, 476, 477, 478, 546, 552, 562, 563, \\n    573, 574, 576, 1001, 1004, 1005, 1006, 1007, 1008, 1009, 1010, 1011, 1012, 1013, \\n    1014, 1015, 1016, 1017, 1018, 1019, 1020, 1021, 1022, 1023, 1024, 1025, 1026, \\n    1028, 1029, 1030, 1031, 1032, 1033, 1036, 1037, 1038, 1039, 1042, 1043, 1046, \\n    1047, 1048, 1049, 1050, 1052, 1053, 1054, 1057, 1058, 1059, 1060, 1061, 1062, \\n    1066, 1067, 1069, 1072, 1091, 1114, 1129, 1133, 1142, 1143, 1149, 1158, 1161, \\n    1164, 1166, 1170, 1175, 1192, 1194, 1199, 1203, 1218, 1219, 1230, 1236, 1239, \\n    1241, 1242, 1243, 1248, 1259, 1261, 1262, 1263, 1264, 1266, 1268, 1371, 1372, \\n    1373, 1375, 1377, 1378, 1494, 1495, 1498, 1502, 1526, 1527, 1529, 1530, 1003, \\n    1051, 1068, 1073, 1079, 1080, 1083, 1084, 1085, 1086, 1088, 1089, 1093, 1096, \\n    1131, 1168, 1177, 1179, 1180, 1190, 1191, 1200, 1237, 1249, 1250, 1507, 1510, \\n    1511, 1512, 1549, 1553, 1557, 1558, 1559, 1560, 1561, 1563, 1564, 1576, 1578, \\n    1593, 1594, 1598, 1613, 1614, 1615, 1617, 1618, 1620, 1621, 1622, 1624, 1625, \\n    1626, 1627, 1629, 1630, 1631, 1632, 1634, 1635, 1786, 1799, 1802, 1804, 1806, \\n    1807, 1808, 1809, 1810, 1811, 1813, 1814, 1816, 1818, 1819, 1825, 1827, 1830, \\n    1835, 1838, 1849, 1852, 1853, 1854, 1855, 1866, 1873, 1908, 1909, 1910, 1911, \\n    1912, 1913, 1915, 1916, 1917, 1918, 1919, 1922, 1924, 1925, 1926, 1927, 1928, \\n    1930, 1931, 1932, 1933, 1936, 1939, 1940, 1941, 1996, 1997, 1998, 2040, 2043, \\n    2044, 2045, 2046, 2047, 2048, 2096, 2106, 2107, 2109, 2110, 2111, 2169, 2170, \\n    2171, 2172\\n]\\n\\n# Create directories if they don’t exist\\nos.makedirs(\"gene_expression_matrices\", exist_ok=True)\\nos.makedirs(\"drug_id_vectors\", exist_ok=True)\\n\\nfor drug_id in drug_ids:\\n    gene_expression_matrix, ic50_vector = create_matrix_and_ic50_for_drug(drug_id, combined_gdsc_df, averaged_data)\\n\\n    if gene_expression_matrix is not None:\\n        gene_expression_matrix.to_csv(f\"gene_expression_matrices/{drug_id}_matrix.csv\")\\n        pd.Series(ic50_vector).to_csv(f\"drug_id_vectors/{drug_id}_vector.csv\", index=False)\\n\\n        print(f\"Saved files for DRUG_ID {drug_id}\")\\n    else:\\n        print(f\"No data for DRUG_ID {drug_id}\")\\n        '"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Export all gene expression matrices and IC50 vectors\n",
    "# Define the list of drug IDs\n",
    "'''\n",
    "drug_ids = [\n",
    "    1, 3, 5, 6, 9, 11, 17, 29, 30, 32, 34, 35, 37, 38, 41, 45, 51, 52, 53, 54, 55, \n",
    "    56, 59, 60, 62, 63, 64, 71, 83, 86, 87, 88, 89, 91, 94, 104, 106, 110, 111, 119, \n",
    "    127, 133, 134, 135, 136, 140, 147, 150, 151, 152, 153, 154, 155, 156, 157, 158, \n",
    "    159, 163, 164, 165, 166, 167, 170, 171, 172, 173, 175, 176, 177, 178, 179, 180, \n",
    "    182, 184, 185, 186, 190, 192, 193, 194, 196, 197, 199, 200, 201, 202, 203, 204, \n",
    "    205, 206, 207, 208, 211, 219, 221, 222, 223, 224, 225, 226, 228, 229, 230, 231, \n",
    "    235, 236, 238, 245, 249, 252, 253, 254, 255, 256, 257, 258, 260, 261, 262, 263, \n",
    "    264, 265, 266, 268, 269, 271, 272, 273, 274, 275, 276, 277, 279, 281, 282, 283, \n",
    "    284, 285, 286, 287, 288, 289, 290, 291, 292, 293, 294, 295, 298, 299, 300, 301, \n",
    "    302, 303, 304, 305, 306, 308, 309, 310, 312, 317, 326, 328, 329, 330, 331, 332, \n",
    "    333, 341, 342, 344, 345, 346, 356, 362, 363, 366, 371, 372, 374, 375, 376, 380, \n",
    "    381, 382, 406, 407, 408, 409, 410, 412, 415, 416, 427, 428, 431, 432, 435, 436, \n",
    "    437, 438, 439, 442, 446, 447, 449, 461, 474, 476, 477, 478, 546, 552, 562, 563, \n",
    "    573, 574, 576, 1001, 1004, 1005, 1006, 1007, 1008, 1009, 1010, 1011, 1012, 1013, \n",
    "    1014, 1015, 1016, 1017, 1018, 1019, 1020, 1021, 1022, 1023, 1024, 1025, 1026, \n",
    "    1028, 1029, 1030, 1031, 1032, 1033, 1036, 1037, 1038, 1039, 1042, 1043, 1046, \n",
    "    1047, 1048, 1049, 1050, 1052, 1053, 1054, 1057, 1058, 1059, 1060, 1061, 1062, \n",
    "    1066, 1067, 1069, 1072, 1091, 1114, 1129, 1133, 1142, 1143, 1149, 1158, 1161, \n",
    "    1164, 1166, 1170, 1175, 1192, 1194, 1199, 1203, 1218, 1219, 1230, 1236, 1239, \n",
    "    1241, 1242, 1243, 1248, 1259, 1261, 1262, 1263, 1264, 1266, 1268, 1371, 1372, \n",
    "    1373, 1375, 1377, 1378, 1494, 1495, 1498, 1502, 1526, 1527, 1529, 1530, 1003, \n",
    "    1051, 1068, 1073, 1079, 1080, 1083, 1084, 1085, 1086, 1088, 1089, 1093, 1096, \n",
    "    1131, 1168, 1177, 1179, 1180, 1190, 1191, 1200, 1237, 1249, 1250, 1507, 1510, \n",
    "    1511, 1512, 1549, 1553, 1557, 1558, 1559, 1560, 1561, 1563, 1564, 1576, 1578, \n",
    "    1593, 1594, 1598, 1613, 1614, 1615, 1617, 1618, 1620, 1621, 1622, 1624, 1625, \n",
    "    1626, 1627, 1629, 1630, 1631, 1632, 1634, 1635, 1786, 1799, 1802, 1804, 1806, \n",
    "    1807, 1808, 1809, 1810, 1811, 1813, 1814, 1816, 1818, 1819, 1825, 1827, 1830, \n",
    "    1835, 1838, 1849, 1852, 1853, 1854, 1855, 1866, 1873, 1908, 1909, 1910, 1911, \n",
    "    1912, 1913, 1915, 1916, 1917, 1918, 1919, 1922, 1924, 1925, 1926, 1927, 1928, \n",
    "    1930, 1931, 1932, 1933, 1936, 1939, 1940, 1941, 1996, 1997, 1998, 2040, 2043, \n",
    "    2044, 2045, 2046, 2047, 2048, 2096, 2106, 2107, 2109, 2110, 2111, 2169, 2170, \n",
    "    2171, 2172\n",
    "]\n",
    "\n",
    "# Create directories if they don’t exist\n",
    "os.makedirs(\"gene_expression_matrices\", exist_ok=True)\n",
    "os.makedirs(\"drug_id_vectors\", exist_ok=True)\n",
    "\n",
    "for drug_id in drug_ids:\n",
    "    gene_expression_matrix, ic50_vector = create_matrix_and_ic50_for_drug(drug_id, combined_gdsc_df, averaged_data)\n",
    "\n",
    "    if gene_expression_matrix is not None:\n",
    "        gene_expression_matrix.to_csv(f\"gene_expression_matrices/{drug_id}_matrix.csv\")\n",
    "        pd.Series(ic50_vector).to_csv(f\"drug_id_vectors/{drug_id}_vector.csv\", index=False)\n",
    "\n",
    "        print(f\"Saved files for DRUG_ID {drug_id}\")\n",
    "    else:\n",
    "        print(f\"No data for DRUG_ID {drug_id}\")\n",
    "        '''\n",
    "\n",
    "#Exporting Drug 1 to .csv\n",
    "\n",
    "# Create a DataFrame for the IC50 row with the same columns as the gene expression data\n",
    "#ic50_row = pd.Series(ic50_vector, index=gene_expression_matrix.columns)\n",
    "\n",
    "# Insert IC50 row at the top of the DataFrame\n",
    "#gene_expression_matrix.loc['IC50'] = ic50_row  # Adding IC50 row\n",
    "\n",
    "# Export to CSV (with IC50 as the first row)\n",
    "#gene_expression_matrix.to_csv('gene_expression_and_ic50.csv', index=True, header=True)\n",
    "\n",
    "# Export to JSON (with IC50 as the first row)\n",
    "#gene_expression_matrix.to_json('gene_expression_and_ic50.json', orient='records', lines=True)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
